packages/vertexai/src/models/imagen-model.ts

```diff
@@ -0,0 +1,199 @@
+  }
+
+  /**
+   * Generates images using the Imagen model and returns them as base64-encoded strings.
+   *
+   * @param prompt The text prompt used to generate the images.
+   * @param imagenRequestOptions Configuration options for the Imagen generation request.
+   * See {@link ImagenGenerationConfig}.
+   * @returns A promise that resolves to an {@link ImagenInlineImageResponse} object containing the generated images.
+   *
+   * @throws If the request fails or if the prompt is blocked, throws a {@link VertexAIError}.
+   *
+   * @remarks
+   * If one or more images are filtered, the returned object will have a defined `filteredReason` property.
+   * If all images are filtered, the `images` array will be empty, and no error will be thrown.
+   */
+  async generateImages(
+    prompt: string,
+    imagenRequestOptions?: ImagenGenerationConfig
+  ): Promise<ImagenInlineImageResponse> {
+    const body = createPredictRequestBody({
+      prompt,
+      ...imagenRequestOptions,
+      ...this.modelConfig
+    });
+    const response = await makeRequest(
+      this.model,
+      Task.PREDICT,
+      this._apiSettings,
+      /* stream */ false,
+      JSON.stringify(body),
+      this.requestOptions
+    );
+    return handlePredictResponse<ImagenInlineImage>(response);
+  }
+
+  /**
+   * Generates images using the Imagen model and returns them as base64-encoded strings.
+   *
+   * @param prompt The text prompt used to generate the images.
+   * @param gcsURI The GCS URI where the images should be stored.
+   * @param imagenRequestOptions Configuration options for the Imagen generation request.
+   * See {@link ImagenGenerationConfig}.
+   * @returns A promise that resolves to an {@link ImagenGCSImageResponse} object containing the generated images.
+   *
+   * @throws If the request fails or if the prompt is blocked, throws a {@link VertexAIError}.
+   *
+   * @remarks
+   * If one or more images are filtered, the returned object will have a defined `filteredReason` property.
+   * If all images are filtered, the `images` array will be empty, and no error will be thrown.
+   */
+  async generateImagesGCS(
+    prompt: string,
+    gcsURI: string,
+    imagenRequestOptions?: ImagenGenerationConfig
+  ): Promise<ImagenGCSImageResponse> {
+    const body = createPredictRequestBody({
+      prompt,
+      gcsURI,
+      ...imagenRequestOptions,
+      ...this.modelConfig
+    });
+    const response = await makeRequest(
+      this.model,
+      Task.PREDICT,
+      this._apiSettings,
+      /* stream */ false,
+      JSON.stringify(body),
+      this.requestOptions
+    );
+    return handlePredictResponse<ImagenGCSImage>(response);
+  }
+}
+
+/**
+ * Creates an {@link ImagenImageFormat} for a JPEG image, to be included in an {@link ImagenModelParams}.
+ *
+ * @param compressionQuality The level of compression.
+ * @returns {@link ImagenImageFormat}
+ *
+ * @public
+ */
+export function jpeg(compressionQuality: number): ImagenImageFormat {
```

Comment:
I'm a little nervous about these names, but I guess it's agreed cross-platform? I'd prefer if it was maybe `jpegFormat()` or namespaced like `ImageFormat.jpeg()` (where ImageFormat is just an object, doesn't have to be a class) but I don't think it's the end of the world.

Reply:
Agreed. I'll make this a static initializer on an `ImageFormat` class.